game-core/src/main/java/games/strategy/triplea/ai/AbstractAi.java

```diff
@@ -236,10 +236,7 @@ public boolean acceptAction(final PlayerID playerSendingProposal, final String a
     final Collection<String> myAlliances =
         new HashSet<>(getGameData().getAllianceTracker().getAlliancesPlayerIsIn(getPlayerId()));
     myAlliances.retainAll(getGameData().getAllianceTracker().getAlliancesPlayerIsIn(playerSendingProposal));
-    if (!myAlliances.isEmpty()) {
-      return true;
-    }
-    return Math.random() < .5;
+    return !myAlliances.isEmpty() || Math.random() < .5;
```

Comment:
General note. We shouldn't rely on Math.random() for anything game-related.
It's not random enough.